### ISSUE TITLE
ActiveStorage images not show in mobile app builtin browser randomly

### ISSUE THREAD

**Issue by yfxie** [rails/rails#35167]
### Steps to reproduce

1. take GCS as ActiveStorage service(not sure whether issue happens for Azure, S3 or disk)
2. upload an image for a record
3. insert the code in view:
```
<%= image_tag record.image %>
<%= image_tag url_for(record.image) %>
<%= image_tag rails_representation_url(record.image.variant(resize: '100').processed) %>
```
4. share the online page to facebook or line message
5. repeatly open and close window by clicking the link in mobile facebook app or line app
6. images will `randomly` not show in the `builtin app browser`

### Expected behavior
all images should show normally for any browser.

### Actual behavior
MacOS: good for desktop browsers: chrome, safari, firefox...
iPhone X: good for the mobile browser: safari, chrome... and images did not show in Facebook/Line App builtin browser randomly.

In the beginning, I thought it's a Facebook App issue. After testing, it also happens in Line builtin browser. And I sure there is `cache-control: max-age=300, private` in redirecting response. and the `Expires` parameter in signed URL is an integer of Time.now + 5min.
so I guessed this issue is because of redirecting. I did a workaround like following:
```
<%= image_tag public_url(record.image) %>
```
the public_url is something like `service.bucket.file(image.key).public_url` with cache for each key to return public_url directly. therefore, browsers never need to ask signed_url first. the issue got solved and got a benefit is a reduction of requesting.
 
In other words, avoid images to take URL from the two controllers https://github.com/rails/rails/blob/v5.2.2/activestorage/app/controllers/active_storage/blobs_controller.rb
https://github.com/rails/rails/blob/v5.2.2/activestorage/app/controllers/active_storage/representations_controller.rb
can solve my issue.

### System configuration
**Rails version**: 5.2.2

**Ruby version**: 2.5.3
